libexec/daps-xmlwellformed

```diff
@@ -212,23 +131,112 @@ def parse_cli(args=None) -> argparse.Namespace:
     return args
 
 
-if __name__ == "__main__":
+def unique(lst: list):
+    """Makes a list unique (preserves order)
+
+    :param lst: the list that needs to be unique
+    :return: the (maybe) reduced list with unique elements
+    """
+    seen = {}
+
+    for item in lst:
+        if item in seen: continue
+        seen[item] = 1
+        yield item
+
+
+def get_all_files(xmlfile: str, xinclude: bool):
+    """Returns a list of all files when enabling XInclude processing
+
+    :param xmlfile: filename of XML file
+    :param xinclude: Enable or disable XInclude processing
+    :return:
+    """
+    ns = etree.FunctionNamespace(SUSE_NS)
+    ns['exists'] = exists
+    ns['abspath'] = abspath
+
+    xifiles = []
+    notfound = []
+    problems = []
+    error = None
+
+    try:
+        tree = etree.parse(xmlfile, parser=XMLPARSER)
+        transform = etree.XSLT(XSLT)
+        if xinclude:
+            result = transform(tree)
+            for entry in transform.error_log:
+                msg = entry.message
+                if msg.startswith("WARN:"):
+                    _, f = msg.split(":")
+                    notfound.append(f)
+                else:
+                    problems.append(str(entry))
+            xifiles = str(result).rstrip().split("\n")
+
+    except ValueError as err:
+        error = err
+        rc = ReturnCode.value_err
+    except etree.XSLTApplyError as err:
+        # We search for "Cannot resolve URI <FILENAME>"
+        uri = err.args[0].split("URI ")[-1]
+        problems.append(uri)
+    except etree.XIncludeError as error:
+        error = err
+        rc = ReturnCode.xinclude_err
+    except etree.XMLSyntaxError as err:
+        error = err
+        rc = ReturnCode.syntax_error
```

Comment:
I had an invalid entity (\&helpsss;) in my MAIN file and got this traceback:

```
  File "../daps/libexec/daps-xmlwellformed", line 165, in get_all_files
    tree = etree.parse(xmlfile, parser=XMLPARSER)
  File "src/lxml/etree.pyx", line 3441, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1834, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1860, in lxml.etree._parseDocumentFromURL
  File "src/lxml/parser.pxi", line 1764, in lxml.etree._parseDocFromFile
  File "src/lxml/parser.pxi", line 1161, in lxml.etree._BaseParser._parseDocFromFile
  File "src/lxml/parser.pxi", line 598, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 709, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 638, in lxml.etree._raiseParseError
  File "xml/MAIN.book-example.xml", line 43
lxml.etree.XMLSyntaxError: Entity 'helpsss' not defined, line 43, column 23

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "../daps/libexec/daps-xmlwellformed", line 242, in <module>
    sys.exit(main())
  File "../daps/libexec/daps-xmlwellformed", line 209, in main
    xifiles, notfound, problems = get_all_files(xmlfile, args.xinclude)
  File "../daps/libexec/daps-xmlwellformed", line 190, in get_all_files
    rc = ReturnCode.syntax_error
NameError: name 'ReturnCode' is not defined
```

Reply:
Argh, seems I left some code which I removed. :scream:  Good catch! Ok, fixed.